AWSPinpoint/AWSPinpointEndpointProfile.m

```diff
@@ -546,4 +561,83 @@ - (void)encodeWithCoder:(NSCoder *)encoder {
     [encoder encodeObject:_userId forKey:@"userId"];
 }
 
+- (void)addUserAttribute:(NSArray *)theValue
+                  forKey:(NSString *)theKey {
+    if (!theKey) return;
+    @synchronized(self) {
+        if (theKey != nil && theValue != nil && ![self hasUserAttributeForKey:theKey]) {
+            if (self.currentNumOfUserAttributes < MAX_NUM_OF_METRICS_AND_ATTRIBUTES) {
+                [self.userAttributes setValue:[AWSPinpointEndpointProfileUser processUserAttributeValues:theValue]
+                                       forKey:[AWSPinpointEndpointProfileUser trimKey:theKey
+                                                                              forType:@"userAttribute"]];
+            } else {
+                AWSDDLogWarn(@"Max number of userAttributes/metrics reached, dropping attribute with key: %@", theKey);
+            }
+        } else if ([self hasUserAttributeForKey:theKey]) {
```

Comment:
seems like this could be an if/else rather than calling `hasUserAttributeForKey` twice?